### PR TITLE
DAOS-9313 test: more log for osa_offline_reintegration_with_rf

### PR DESCRIPTION
**Pull request by Nasf-Fan**
Enable "io,trace" in the DD_MASK.

Test-tag: offline_reintegration_with_rf online_parallel
Test-repeat-hw-medium: 30

Signed-off-by: Fan Yong <fan.yong@intel.com>

### CODE REVIEW

**Comment by Nasf-Fan** [comment]
We hope to reproduce the issue, then restart the test.

**Comment by Nasf-Fan** [comment]
Set 12 failed on osa_offline_reintegration again, but this time, it is caused by DAOS-9975. Then have to restart the test for further try.

File: src/tests/ftest/osa/offline_reintegration.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ server_config:
       scm_mount: /mnt/daos0
       log_mask: DEBUG,MEM=ERR
       env_vars:
-        - DD_MASK=mgmt,md,rebuild
+        - DD_MASK=mgmt,md,rebuild,io,trace
```
[the remark this answers is not in the document]
Comment (Nasf-Fan) [reply src/tests/ftest/osa/offline_reintegration.yaml:31]: I repeated the test many times with the patch, nerve reproduce the issue, then I have to consider to land it to master, once we fix related issue, this patch can be rollback together the fixes.

**Comment by Nasf-Fan** [comment]
We need to land this patch to collective more logs.

**Comment by jolivier23** [comment]
> We need to land this patch to collect more logs.

how often does the issue happen?   Could we set the commit pragma to run the test(s) in question 100 times or some such and go from there?   Or is it so rare that it needs to be landed?

**Review by jolivier23 — CHANGES_REQUESTED** [review]
Adding  a review to remove from inbox while question answered.

**Comment by Nasf-Fan** [comment]
> I think what you would want is something like
> Test-tag: [identifying tag for the failing test(s)]
> Test-repeat: 100

Here is some commit parameters for the PR:

Test-tag-hw-medium: osa_offline_reintegration
Test-repeat: 30

But osa_offline_reintegration did not run in CI test. Is there any wrong?
@brianjmurrell , thanks!

**Comment by brianjmurrell** [comment]
> Here is some commit parameters for the PR:
> 
> Test-tag-hw-medium: osa_offline_reintegration
> Test-repeat: 30
> 
> But osa_offline_reintegration did not run in CI test. Is there any wrong? @brianjmurrell , thanks!

Could be [DAOS-10029](https://daosio.atlassian.net/browse/DAOS-10029).  @phender Thoughts?

**Comment by phender** [comment]
> > Here is some commit parameters for the PR:
> > Test-tag-hw-medium: osa_offline_reintegration
> > Test-repeat: 30
> > But osa_offline_reintegration did not run in CI test. Is there any wrong? @brianjmurrell , thanks!
> 
> Could be [DAOS-10029](https://daosio.atlassian.net/browse/DAOS-10029). @phender Thoughts?
> 
> As an aside @phender, because the pragma used was `Test-tag-hw-medium` and not `Test-tag` would that mean that all test stages (i.e. VMs, hardware small and large) would have run all of their tests 30x, unnecessarily tying up (hardware) clusters for an extended amount of time? If so, should we perhaps only allow `Test-repeat` to be used with `Test-tag` and not any of the `Test-tag-<stage>` pragmas? At least then, the user gets an error message up front instead of unintentionally tying up valuable hardware resources for long periods of time. Or the framework could just internally drop any `-<stage>` from `Test-tag-<stage>` to effect what the user most likely actually wanted.

The first problem is `osa_offline_reintegration` is not a test tag.  The [`osa/offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegration_with_rf`](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/osa/offline_reintegration.py#L289-L293) test uses these test tags:
```
        :avocado: tags=all,daily_regression
        :avocado: tags=hw,medium,ib2
        :avocado: tags=osa,checksum
        :avocado: tags=offline_reintegration_full,mpich
        :avocado: tags=offline_reintegration_with_rf
```
The `offline_reintegration_with_rf` tag is unique and the one I'd suggest to use.  Then you'll run into DAOS-10029 because this test uses python3-specific code, so you'll need to use an additional test tag (one for a test is currently skipped) to get past the `list_tests.py` check that verifies if the stage has any tests to run.  In this case I'd suggest using `online_parallel`.

So to run just the `osa/offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegration_with_rf` test use either:
```
Test-tag: offline_reintegration_with_rf online_parallel
```
or:
```
Test-tag-hw-medium: offline_reintegration_with_rf online_parallel
```

The second - and major - issue is the pragmas being used are incomplete to achieve the desired goal of running one test multiple times.  The problem with:
```
Test-tag-hw-medium: offline_reintegration_with_rf online_parallel
Test-repeat: 30
```
Is that it only reduces the tests run in the HW Medium stage and will still run the other functional test stages with the `pr` tag (the default) so all the tests in `Functional on EL 8`, `Functional on HW Small`, and `Functional on HW Large` are run 30 times - not good for CI at all.

We can limit the test repetition by either explicitly targeting the stage:
```
Test-repeat-hw-medium: 30
``` 
or eliminating any tests from being run on the other functional test stages indirectly with:
```
Test-tag: offline_reintegration_with_rf online_parallel
```
or directly with:
```
Skip-unit-tests: true
Skip-func-test-vm: true
Skip-func-hw-test-small: true
Skip-func-hw-test-large: true
Test-tag-hw-medium: offline_reintegration_with_rf online_parallel
```

So putting it all together these are the tags that should be used for this PR:
```
Test-tag: offline_reintegration_with_rf online_parallel
Test-repeat-hw-medium: 30
```

**Comment by Nasf-Fan** [comment]
Thanks @brianjmurrell @phender !
Let me try again with the new parameters.

**Comment by Nasf-Fan** [comment]
> So putting it all together these are the pragmas that should be used for this PR:

> Test-tag: offline_reintegration_with_rf online_parallel
> Test-repeat-hw-medium: 30

The PR has been refreshed for more than two weeks, but the expected test seemed still not run. @brianjmurrell

**Comment by daltonbohning** [comment]
> > So putting it all together these are the pragmas that should be used for this PR:
> 
> > Test-tag: offline_reintegration_with_rf online_parallel
> > Test-repeat-hw-medium: 30
> 
> The PR has been refreshed for more than two weeks, but the expected test seemed still not run. @brianjmurrell

The test was in the skiplist:
`[['DAOS-10435', 'test_method_name', 'test_osa_offline_reintegration_with_rf']]|0ccff36`
And commit `0ccff36` was merged after this PR's base. So I think it should work if you merge master and re-run

**Comment by Nasf-Fan** [comment]
Not need it any longer.
